extensions/wikia/Chat2/ChatEntryPoint.class.php

```diff
@@ -23,35 +23,56 @@ static public function onParserFirstCallInit( Parser &$parser ) {
 	}
 
 	/**
-	 * @param $isEntryPoint - set to true for chat entry point embeded in the article, false for rail module
-	 * Return an array of variables needed to render chat entry point template
+	 * @param $isEntryPoint - set to true for chat entry point embedded in the article,
+	 * set to false for rail module
+	 *
+	 * Return an array of variables needed to render chat entry point mustache template
 	 *
 	 * @return array
 	 */
 	static public function getEntryPointTemplateVars( $isEntryPoint ) {
-		global $wgEnableWallExt, $wgBlankImgUrl;
-		return [
-			'linkToSpecialChat' => SpecialPage::getTitleFor( "Chat" )->escapeLocalUrl(),
-			'isEntryPoint' => $isEntryPoint,
+		global $wgEnableWallExt, $wgBlankImgUrl, $wgUser, $wgSitename;
+
+		$entryPointGuidelinesMessage = wfMessage( 'chat-entry-point-guidelines' );
+		$joinTheChatMessage = wfMessage( 'chat-join-the-chat' );
+		$chatUsers = ChatEntryPoint::getChatUsersInfo();
```

Comment:
I like it to be a bit more specific here and name it `chatUserInfos` instead of jsut plain `chatUsers` as this may suggest to be a set of `User` objects.